### PR TITLE
Introduce pluggable event modifiers

### PR DESCRIPTION
**Pull request by bernd**
This replaces specific risk score calculators with a pluggable event modifiers system.

Refs #18040

/nocl

### CODE REVIEW

File: graylog2-server/src/main/java/org/graylog2/plugin/PluginModule.java [hunk 1]
```diff
@@ -246,7 +246,16 @@ protected void addEventProcessor(String name,
...
+    protected Multibinder<EventModifier> eventModifierBinder() {
+        return Multibinder.newSetBinder(binder(), EventModifier.class);
+    }
+
+    protected void addEventModifier(Class<? extends EventModifier> eventModifierClass) {
+        eventModifierBinder().addBinding().to(eventModifierClass);
+    }
+
     private MapBinder<String, EventStorageHandler.Factory> eventStorageHandlerBinder() {
+        Multibinder.newSetBinder(binder(), SchedulerCapabilities.class);
```
Comment (kingzacko1) [inline graylog2-server/src/main/java/org/graylog2/plugin/PluginModule.java:258]: was this supposed to be added with these changes?

Comment (bernd) [reply graylog2-server/src/main/java/org/graylog2/plugin/PluginModule.java:258]: Oups, no. Good catch, thank you! I will fix it.

**Review by kingzacko1 — APPROVED** [review]
LGTM!

**Review by danotorrey — APPROVED** [review]

**Comment by bernd** [comment]
@kingzacko1 @danotorrey Re-requesting review because I pushed more changes after your approvals.

**Review by kingzacko1 — APPROVED** [review]
All LGTM!

**Pull request merged**